### PR TITLE
Merge Data and Meta Roles

### PR DESCRIPTION
**Pull request by pmyjavec**
This change makes it easier to deploy and test the Ansible enterprise role by merging the data and meta roles.

* Ensure all params are set in the template / mandatory
* Merge tasks, vars handlers etc
* Re-order the repo structure, roles at the top level
* Simplify Vagrant box slightly

### CODE REVIEW

**Comment by kostasb** [comment]
We can use the latest packages as well:

https://s3.amazonaws.com/dl.influxdata.com/enterprise/releases/influxdb-meta_1.2.1-c1.2.2_amd64.deb

https://s3.amazonaws.com/dl.influxdata.com/enterprise/releases/influxdb-meta_1.2.1-c1.2.2_amd64.deb

File: influxdb-enterprise/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,106 @@
+---
+# Role Settings
+influx_node_types: ["meta", "data"]
+influx_node_type: "meta" # By default, install node as a meta-node
+influx_pkg_base_url: "https://s3.amazonaws.com/dl.influxdata.com/enterprise/releases"
+influx_version: "1.2.1-c1.2.2" # The version of the package to install
```
Comment (pmyjavec) [inline influxdb-enterprise/defaults/main.yml:6]: @kostasb, I took your suggestion on-board and implemented this. Now both packages are installed at the exact same versions.

**Comment by pmyjavec** [comment]
This is totally working now, it *will* build a new cluster from scratch. The problem I was having with the data-nodes was that I was trying to add them to the meta nodes using the `HTTP` port and not the `RPC` port. I've talked to Reagan about updating the docs for that.

**Review by kostasb — APPROVED** [review]
Let's go ahead and merge

**Pull request merged**